process_request/helpers.py

```diff
@@ -110,11 +174,24 @@ def is_restricted(archival_object, query_string, restriction_acts):
     :returns: True if archival object is restricted, False if not.
     :rtype: bool
     """
-    for note in archival_object["notes"]:
-        if note["type"] == "accessrestrict":
+    for note in archival_object.get("notes"):
+        if note.get("type") == "accessrestrict":
             if text_in_note(note, query_string.lower()):
                 return True
-    for rights_statement in archival_object["rights_statements"]:
+    for rights_statement in archival_object.get("rights_statements"):
         if indicates_restriction(rights_statement, restriction_acts):
             return True
     return False
+
+
+def object_locations(archival_object):
+    """Finds locations associated with an archival object.
+    :param JSONModelObject archival_object: an ArchivesSpace archival_object.
+    :returns: Locations objects associated with the archival object.
+    :rtype: list
+    """
+    locations = []
+    for instance in archival_object.instances:
+        top_container = instance.sub_container.top_container.reify()
+        locations += top_container.container_locations
+    return locations
```

Comment:
This method does not appear to be called; can we delete it?

Reply:
I left this alone since I didn't know if it would be helpful for @p-galligan's work.

Reply:
I think we can delete it. I can always add it back if it becomes necessary. However, if we're using the resolve approach, I don't think it will be.

Reply:
There's already a `get_locations` method that, based on the docstrings, should do the same thing as this, and works with resolved data rather than making additional fetches.